connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTaskTest.java

```diff
@@ -78,155 +74,80 @@ public void tearDown() {
-
-        WorkerTask workerTask = partialMockBuilder(WorkerTask.class)
-                .withConstructor(
-                        ConnectorTaskId.class,
-                        TaskStatus.Listener.class,
-                        TargetState.class,
-                        ClassLoader.class,
-                        ConnectMetrics.class,
-                        RetryWithToleranceOperator.class,
-                        Time.class,
-                        StatusBackingStore.class
-                )
-                .withArgs(taskId, statusListener, TargetState.STARTED, loader, metrics,
-                        retryWithToleranceOperator, Time.SYSTEM, statusBackingStore)
-                .addMockedMethod("initialize")
-                .addMockedMethod("initializeAndStart")
-                .addMockedMethod("execute")
-                .addMockedMethod("close")
-                .createStrictMock();
-
         final CountDownLatch stopped = new CountDownLatch(1);
-        final Thread thread = new Thread(() -> {
-            try {
-                stopped.await();
-            } catch (Exception e) {
-            }
-        });
-
-        workerTask.initialize(TASK_CONFIG);
-        EasyMock.expectLastCall();
-
-        workerTask.initializeAndStart();
-        EasyMock.expectLastCall();
-
-        workerTask.execute();
-        expectLastCall().andAnswer(() -> {
-            thread.start();
-            return null;
-        });
 
-        statusListener.onStartup(taskId);
-        expectLastCall();
+        WorkerTask workerTask = new TestWorkerTask(taskId, statusListener, TargetState.STARTED, loader, metrics,
+                retryWithToleranceOperator, Time.SYSTEM, statusBackingStore) {
 
-        workerTask.close();
-        expectLastCall();
-
-        // there should be no call to onShutdown()
+            @Override
+            public void execute() {
+                try {
+                    stopped.await();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
 
-        replay(workerTask);
+            // Trigger task shutdown immediately after start. The task will block in it's execute() method
+            // until the stopped latch is counted down (i.e. it doesn't actually stop after stop is triggered).
+            @Override
+            public void initializeAndStart() {
+                stop();
+            }
+        };
 
         workerTask.initialize(TASK_CONFIG);
-        workerTask.run();
+        Thread t = new Thread(workerTask);
+        t.start();
 
-        workerTask.stop();
         workerTask.cancel();
         stopped.countDown();
-        thread.join();
+        t.join();
 
-        verify(workerTask);
+        verify(statusListener).onStartup(eq(taskId));
+        // there should be no call to onShutdown()
+        verify(statusListener, never()).onShutdown(eq(taskId));
```

Comment:
There should be no more interactions whatsoever, right? Probably makes more sense to verify that than to just verify that we don't record a shutdown event:
```suggestion
        // there should be no other status updates, including shutdown
        verifyNoMoreInteractions(statusListener);
```